hub/core/tensor.py

```diff
@@ -131,10 +144,18 @@ def read_samples_from_tensor(
     meta = read_tensor_meta(key, storage)
     index_map = read_index_map(key, storage)
 
+    dtype = meta["dtype"]
+
     # TODO: read samples in parallel
     samples = []
     for index_entry in index_map[index.to_slice()]:
-        array = sample_from_index_entry(key, storage, index_entry, meta["dtype"])
+        shape = index_entry["shape"]
+
+        if is_shape_empty(shape):
+            samples.append(np.zeros(shape, dtype=dtype))
+            break  # no need to actually load data (no chunks exist)
```

Comment:
samples may contain `0` in their shape, this is for the reasons mentioned before (some image samples may have 0 corresponding BBoxes, in this case it makes sense to store a sample with a shape like: `(0, 4)` (0 bboxes)

Reply:
@McCrearyD didn't I see that we're throwing an exception if there's a 0 in the shape?

Reply:
yeah this is an outdated comment, i was going to implement it but there were a lot of things that came up so it will have to be a separate PR